Enable Timelines from accordion on Storage Managers page

hstastna opened the pull request:
fixing https://bugzilla.redhat.com/show_bug.cgi?id=1383388

Make Timelines clickable from accordion on Storage Managers
summary page when navigating to Storage -> Storage Managers
and choosing a storage manager from the list
because Timelines was broken and working only in monitoring tab.

The condition on
https://github.com/hstastna/manageiq/blob/master/app/views/layouts/listnav/_ems_storage.html.haml#L13 
was changed because it makes no sense to disable
Timelines page from accordion and enable the page only if
`(@record.has_events? || @record.has_events?(:policy_events))`.
In monitoring tab, Timelines page is enabled even if the condition above is false.

Before:
![timelines_before](https://cloud.githubusercontent.com/assets/13417815/20268528/d5a301d2-aa7f-11e6-93a8-c3436d4f077a.png)

After:
![timelines_after](https://cloud.githubusercontent.com/assets/13417815/20268660/64a11388-aa80-11e6-9f95-33760d7d2556.png)
![timelines_after2](https://cloud.githubusercontent.com/assets/13417815/20268667/67af3ee2-aa80-11e6-8cfe-2155a56650bf.png)

hstastna commented:
@miq-bot add_label bug, ui

dclarizio commented:
@h-kataria can you review this?  I thought we didn't enable timelines button or listnav link if there were no records to show.  Thx, Dan

h-kataria commented on app/views/layouts/listnav/_ems_storage.html.haml:
```diff
@@ -6,13 +6,13 @@
           :record_id  => @record.id,
           :display    => 'main',
           :title      => _("Show Summary"))
-        = li_link(:if => (@record.has_events? || @record.has_events?(:policy_events)),
```
@hstastna we need to keep this check, Timelines link is only available when there is timeline data available for either Policy or Management Events.

hstastna replied:
@h-kataria Does it mean that Timelines is not broken from accordion on Storage Managers summary page but the bug is that it is enabled in Monitoring tab even there is no timeline data?

h-kataria replied:
@hstastna yes, both link & button should check for events availability. Need to add a check similar to a check for Timeline button on Cluster summary screen https://github.com/ManageIQ/manageiq/blob/master/app/helpers/application_helper/toolbar_builder.rb#L707

hstastna commented:
According to @h-kataria 's comments, I added check for timeline data available for Policy or Management Events.
I also added (in separate commit) title for Timelines in accordion when it is disabled to reach more consistent behaviour.

simaishi commented:
Euwe backport details:

```
$ git log -1
commit 2132f57b2c30caccac4481098a3ae956749abbb0
Author: Harpreet Kataria <hkataria@redhat.com>
Date:   Mon Nov 28 15:49:27 2016 -0500

    Merge pull request #12613 from hstastna/TimeLine_accordion_on_storage_managers
    
    Enable Timelines from accordion on Storage Managers page
    (cherry picked from commit 6d4cf75b84c84fb5dc55fd39508b7ecaf2729015)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1411461
```
